qiskit/algorithms/optimizers/fraxis.py

```diff
@@ -0,0 +1,202 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Free-Axis Selection (Fraxis) algorithm."""
+
+from typing import Optional, Tuple
+
+import numpy as np
+from scipy.optimize import OptimizeResult
+
+from qiskit.circuit.library import UGate
+from qiskit.quantum_info import OneQubitEulerDecomposer, Pauli
+
+from .scipy_optimizer import SciPyOptimizer
+
+
+class Fraxis(SciPyOptimizer):
+    """
+    Free-Axis Selection (Fraxis) algorithm [1].
+
+    More precisely, this class implements π-Fraxis algorithm in Algorithm 1 of [1].
```

Comment:
Do you think we could put some sort or overview/summary here in the docs so a user coming to it has an idea what this is and when perhaps to use it. They may want to go read the reference paper but some description here maybe helpful for people that do not want to delve into detail of that paper. Maybe we can note the 6 evals per iter here too.